### DISCUSSION TITLE
Redeploy using Vercel API

### DISCUSSION THREAD

**Discussion by KYesmine** [vercel/community#3140] [Help] [answered]
### Summary

Hey everyone,

I'm trying to find out if there is an endpoint in the Vercel API that allows for redeploying a project. I've gone through the documentation, but I couldn't find any specific endpoint for this purpose. However, during my research, ChatGPT provided me with the following two endpoints:

https://api.vercel.com/v2/now/deployments/{deployment_id}/redeploy
https://api.vercel.com/v12/now/deployments/{deployment_id}/redeploy

I attempted to use these endpoints, but I encountered an error consistently when testing them with my different project. The error message I received was as follows:

'Infinite loop detected
INFINITE_LOOP_DETECTED'

I would appreciate it if someone could confirm whether these endpoints are correct or provide any guidance on how to redeploy a project using the Vercel API. Thank you in advance for your help!

### Steps to Reproduce

No steps needed

**Comment by amyegan** [comment, marked as answer]
Hi @KYesmine. The endpoints provided by ChatGPT are not correct. You can use `POST` to `https://api.vercel.com/v13/deployments` with the id of a previous deployment to redeploy it. The documentation, with examples, is here: https://vercel.com/docs/rest-api/endpoints#create-a-new-deployment

**Comment by chenhanting-ibuypower** [comment]
````sh
curl --location 'https://api.vercel.com/v13/deployments' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer <My Token>' \
--data '{
    "name": "my-instant-deployment",
    "deploymentId": "<DeployID>"
}'
````

Is like this?

What if i want to deploy with the specific branch via RESTful API
